.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,19 @@
+{
+	"name": "Immich devcontainers",
+	"build": {
+		"dockerfile": "Dockerfile",
+		"args": {
+			"BASEIMAGE": "mcr.microsoft.com/devcontainers/typescript-node:22"
+		}
```

Comment:
Why is this defined as an arg, and in two separate places?

Reply:
No, I have no intention of adding anything else to this PR.
It is just an "initial" version of a working devcontainer that makes it easy to contribute to some components of immich (web, doc...) without having to setup a development environment. I'll let other people contribute !

Regarding `BASEIMAGE`, it was inspired by other open source projects => most likely it is easier to update the base image from the json file (rather than editing Dockerfile).
Also, for now, the Dockerfile is empty (just the base image), but it will allow to easily add software & tools later.